src/VisualStudio/Core/Def/Implementation/InheritanceMargin/InheritanceGlyphManager.cs

```diff
@@ -0,0 +1,210 @@
+                var glyphToTaggedSpanBuilder = new Dictionary<InheritanceMarginGlyph, SnapshotSpan>(_glyphToTaggedSpan.Count);
+
+                foreach (var (glyph, span) in _glyphToTaggedSpan)
+                {
+                    var newSpan = span.TranslateTo(snapshot, SpanTrackingMode.EdgeInclusive);
+                    if (!_textView.TextViewLines.IntersectsBufferSpan(newSpan) || (GetStartingLine(newOrReformattedLines, span) != null))
+                    {
+                        //Either visual is no longer visible or it crosses a line
+                        //that was reformatted.
+                        _glyphsContainer.Children.Remove(glyph);
+                    }
+                    else
+                    {
+                        glyphToTaggedSpanBuilder[glyph] = newSpan;
+                        var line = GetStartingLine(translatedLines, span);
+                        if (line != null)
+                        {
+                            SetTop(line, glyph);
+                        }
+                    }
+                }
+
+                _glyphToTaggedSpan = glyphToTaggedSpanBuilder;
+            }
+        }
+
+        private void SetTop(ITextViewLine line, InheritanceMarginGlyph glyph)
+            => Canvas.SetTop(glyph, line.TextTop - _textView.ViewportTop);
+
+        private static ITextViewLine? GetStartingLine(IList<ITextViewLine> lines, Span span)
+        {
+            if (lines.Count > 0)
+            {
+                var low = 0;
+                var high = lines.Count;
+                while (low < high)
+                {
+                    var middle = (low + high) / 2;
+                    var middleLine = lines[middle];
+                    if (span.Start < middleLine.Start)
+                        high = middle;
+                    else if (span.Start >= middleLine.EndIncludingLineBreak)
+                        low = middle + 1;
+                    else
+                        return middleLine;
+                }
+
+                var lastLine = lines[lines.Count - 1];
+                if ((lastLine.EndIncludingLineBreak == lastLine.Snapshot.Length) && (span.Start == lastLine.EndIncludingLineBreak))
+                {
+                    // As a special case, if the last line ends at the end of the buffer and the span starts at the end of the buffer
+                    // as well, treat is as crossing the last line in the buffer.
+                    return lastLine;
+                }
+            }
+
+            return null;
+        }
+
+        private InheritanceMarginGlyph CreateNewGlyph(InheritanceMarginTag tag)
+            => new InheritanceMarginGlyph(
+                _threadingContext,
+                _streamingFindUsagesPresenter,
+                _classificationTypeMap,
+                _classificationFormatMap,
+                _operationExecutor,
+                tag,
+                _textView,
+                _listener);
```

Comment:
consider having the glyph just point at this manager, and allow it to grab these fields off of it.  up to you though!